UploadBehavior.php

```diff
@@ -197,9 +197,8 @@ public function afterSave()
      */
     public function beforeDelete()
     {
-        $attribute = $this->attribute;
-        if ($this->unlinkOnDelete && $attribute) {
-            $this->delete($attribute);
+        if ($this->unlinkOnDelete) {
+            $this->deleteFolder($this->path);
```

Comment:
If I have right understanding we have just one place where we must delete files and leave folder - beforeSave() in case if 'unlinkOnSave' == true. As for me we don't have to check that directory is empty if we want to completely remove model with all it's files.